azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,476 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vm_size": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "Standard_A1",
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ForceNew:         true,
+			},
+			"scale_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(azure.BatchPoolFixedScale),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(azure.BatchPoolFixedScale),
+					string(azure.BatchPoolAutoScale),
+				}, false),
+			},
+			"target_dedicated_nodes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"target_low_priority_nodes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"resize_timeout": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "PT15M",
+			},
+			"autoscale_evaluation_interval": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "PT15M",
+			},
+			"autoscale_formula": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"storage_image_reference": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"publisher": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: resourceArmVirtualMachineStorageImageReferenceHash,
+			},
+			"node_agent_sku_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"stop_pending_resize_operation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchPoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Batch pool creation.")
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	accountName := d.Get("account_name").(string)
+	name := d.Get("name").(string)
+	displayName := d.Get("display_name").(string)
+	vmSize := d.Get("vm_size").(string)
+
+	parameters := batch.Pool{
+		PoolProperties: &batch.PoolProperties{
+			VMSize:      &vmSize,
+			DisplayName: &displayName,
+		},
+	}
+
+	scaleMode := azure.BatchPoolScaleMode(d.Get("scale_mode").(string))
+
+	if scaleMode == azure.BatchPoolAutoScale {
```

Comment:
I think it would be best to put these properties into a `fixed_scale` and `auto_scale` blocks that conflict with each other, and they we can drop the comparison here.

Reply:
Done.